lib/virt_autotest/utils.pm

```diff
@@ -42,39 +42,58 @@ sub trim {
     return $text;
 }
 
-sub restart_libvirtd {
-    if (is_sle('<12')) {
-        assert_script_run('rclibvirtd restart', 180);
-    }
-    elsif (is_alp) {
-        my $_libvirtd_pid = script_output(q@ps -ef |grep [l]ibvirtd | gawk '{print $2;}'@);
-        my $_libvirtd_cmd = script_output("ps -o command $_libvirtd_pid | tail -1");
-        assert_script_run("kill -9 $_libvirtd_pid");
-        assert_script_run("$_libvirtd_cmd");
-    }
-    else {
-        systemctl("restart libvirtd", timeout => 180);
-    }
-    record_info("Libvirtd has been restarted!");
-    save_screenshot;
-}
-
 #return 1 if test is expected to run on XEN hypervisor
 sub is_xen_host {
     return get_var("XEN") || check_var("SYSTEM_ROLE", "xen") || check_var("HOST_HYPERVISOR", "xen") || check_var("REGRESSION", "xen-hypervisor");
 }
 
+#Setup modular libvrit daemons as the default in SLE15SP6, refer to PED-4804 to get more details
+sub is_modular_libvirtd {
+    return 0 if is_sle(">=15-sp6");
+}
```

Comment:
there is `is_monolithic_libvirtd` function in `lib/virt_autotest/utils.pm`. It can be used to differentiate modular libvirt or monolithic libvirtd.

Reply:
Good point. Currently, set up the modular libvirt daemons as the default on ALP and openSUSE TW, and next, plan to set up modular libvirt in SLES15SP6. So, we'd better create is_modular_libvirtd() to deal with the different situations on ALP, OpenSUSE TW, and SLES15SP6 about the modular libvirt daemons

Reply:
Sorry I don't quite understand your point. Do you mean that is_modular_libvirtd() is still required even with current is_monolithic_libvirtd()? or you need assert all the modular daemons are running as expected on a modular libvirt system?

Reply:
No, I means that in the future, we will support modular libvirt daemons as the default, but in the meantime, there may be inconsistencies in the support for openSUSE tw, alp, and sles15sp6. Using this existing is_monolithic_libvirtd() is not enough for the future modular libvirt daemons scenarios.

For example:
```
sub is_modular_libvirtd {
    if is_sle(">=15-sp6") {};
    elseif is_alp() {}
    elseif is_tw() {}
    else .... 
}
```